carbon-design-system/carbon
# fix(TableToolbarSearch): allow default values to be set

Closes https://github.com/carbon-design-system/carbon/issues/3170

The `DefaultValue` prop was being ignored. This checks for the presence of a `DefaultValue`, and if it exists, it is used as the default state of `value`. On the initial load, it sets the expanded state to `true` and runs the `onChangeProp` to filter the table. Modified the `handleOnInputValueChange` in `DataTable.js` to account for non-event based sorting (table loads with a default value). 

<img width="1012" alt="Screen Shot 2020-01-17 at 2 10 40 PM" src="https://user-images.githubusercontent.com/11928039/72649771-6216d400-3933-11ea-89ee-5cb413af0b6f.png">


#### Changelog

**New**

- `DefaultValue` prop works as intended

**Changed**

- `handleOnInputValueChange` now has a `defaultValue` parameter

**Removed**

- `DefaultValue` being passed directly down to `Search`. If `DefaultValue` exists, it is immediately set to `value` which is then passed down. 

#### Testing / Reviewing

- Open `src/components/DataTable/stories/with-toolbar.js`
- Add `defaultValue="1"` to `TableToolbarSearch`
- Open the `With Toolbar` Storybook example, verify only 1 result shows up.
- Remove the value, verify all results show up.
- Add new value and verify the correct results show up.
